servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncPublisherOperator.java

```diff
@@ -357,8 +383,12 @@ private void enqueueTaskIfRequired(boolean forRequestN) {
             final int oldState = stateUpdater.getAndSet(this, STATE_ENQUEUED);
             if (oldState == STATE_IDLE) {
                 try {
-                    LOGGER.debug("executing {} on {}", forRequestN ? "request" : "cancel", executor);
-                    executor.execute(this::executeTask);
+                    if (offload.getAsBoolean()) {
+                        LOGGER.debug("executing {} on {}", forRequestN ? "request" : "cancel", executor);
```

Comment:
remove debug/development logs? operators allow for instrumentation/visibility if required.